src/Nethermind/Nethermind.Core/Crypto/Keccak.cs

```diff
@@ -5,22 +5,93 @@
+        public Span<byte> BytesAsSpan => MemoryMarshal.AsBytes(MemoryMarshal.CreateSpan(ref Unsafe.AsRef(in Bytes), 1));
+
+        public ReadOnlySpan<byte> Span => MemoryMarshal.AsBytes(MemoryMarshal.CreateReadOnlySpan(ref Unsafe.AsRef(in Bytes), 1));
 
         /// <returns>
         ///     <string>0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470</string>
         /// </returns>
         public static readonly ValueKeccak OfAnEmptyString = InternalCompute(new byte[] { });
 
+        /// <returns>
+        ///     <string>0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347</string>
+        /// </returns>
+        public static readonly ValueKeccak OfAnEmptySequenceRlp = InternalCompute(new byte[] { 192 });
+
+        /// <summary>
+        ///     0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421
+        /// </summary>
+        public static readonly ValueKeccak EmptyTreeHash = InternalCompute(new byte[] { 128 });
+
+        /// <returns>
+        ///     <string>0x0000000000000000000000000000000000000000000000000000000000000000</string>
+        /// </returns>
+        public static ValueKeccak Zero { get; } = default;
+
+        /// <summary>
+        ///     <string>0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff</string>
+        /// </summary>
+        public static ValueKeccak MaxValue { get; } = new("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
+
+        public static implicit operator ValueKeccak(Keccak? keccak)
+        {
+            return new ValueKeccak(keccak?.Bytes);
+        }
+
+        public ValueKeccak(byte[]? bytes)
+        {
+            if (bytes is null || bytes.Length == 0)
+            {
+                Bytes = OfAnEmptyString.Bytes;
+                return;
+            }
+
+            Debug.Assert(bytes.Length == ValueKeccak.MemorySize);
+            Bytes = Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetArrayDataReference(bytes));
+        }
+
+        public ValueKeccak(string? hex)
+        {
+            if (hex is null || hex.Length == 0)
+            {
+                Bytes = OfAnEmptyString.Bytes;
+                return;
+            }
+
+            byte[] bytes = Extensions.Bytes.FromHexString(hex);
+            Debug.Assert(bytes.Length == ValueKeccak.MemorySize);
+            Bytes = Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetArrayDataReference(bytes));
+        }
+
+        public ValueKeccak(Span<byte> bytes)
+            : this((ReadOnlySpan<byte>)bytes) { }
+
+        public ValueKeccak(ReadOnlySpan<byte> bytes)
+        {
+            if (bytes.Length == 0)
+            {
+                Bytes = OfAnEmptyString.Bytes;
+                return;
+            }
+
+            Debug.Assert(bytes.Length == ValueKeccak.MemorySize);
+            Bytes = Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(bytes));
```

Comment:
same question about span

Reply:
Is a by value copy as `Bytes` is a `Vector256<byte>` type and doesn't hold a reference to anything